### PR TITLE
Adapts rekeying-function to ksqlDB 0.7.1

### PR DESCRIPTION
**Pull request by colinhicks**

### CODE REVIEW

File: _includes/tutorials/rekeying-function/ksql/code/tutorial-steps/dev/continuous-regex.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CREATE STREAM customers_with_area_code AS
+    SELECT 
+      customers.rowkey as id,
+      firstname,
+      lastname,
+      phonenumber,
+      REGEXREPLACE(phonenumber, '\\(?(\\d{3}).*', '$1') as area_code
+    FROM customers;
```
Comment (colinhicks) [inline _includes/tutorials/rekeying-function/ksql/code/tutorial-steps/dev/continuous-regex.sql]: this intermediate stream is necessary in 0.7.0+, when expression support was added to `PARTITION BY` 

cf. https://github.com/confluentinc/ksql/pull/4032

**Comment by rspurgeon** [comment]
Not certain what causes this difference, but when I run the DESCRIBE FUNCTION REGEXREPLACE, I get a different jar path:

```
ksql> DESCRIBE FUNCTION REGEXREPLACE;

Name        : REGEXREPLACE
Overview    : Replace string using a regex
Type        : SCALAR
Jar         : /etc/ksql/ext/code-0.0.1.jar
Variations  :

        Variation   : REGEXREPLACE(input VARCHAR, regex VARCHAR, replacement VARCHAR)
        Returns     : VARCHAR
        Description : regexReplace string using a regex
        input       : If null, then function returns null.
        regex       : If null, then function returns null.
        replacement : If null, then function returns null.
```

**Comment by colinhicks** [comment]
Yeah, that's interesting. I'm not sure how to account for the jar name difference either. I think it's innocuous though, in this case.

**Comment by rspurgeon** [comment]
@colinhicks this tutorial references `compile 'io.confluent.ksql:ksql-udf:5.3.0'`

**Pull request merged**